battstat/battstat_applet.c

```diff
@@ -108,648 +110,662 @@ status_change_callback (void)
-
-  if (info->minutes < 0 && !info->on_ac_power)
-  {
-	  /* we don't know the remaining time */
-	  remaining = g_strdup_printf (_("You have %d%% of your total battery "
-				  	 "capacity remaining."), info->percent);
-  }
-  else
-  {
-	  remaining = g_strdup_printf( ngettext(
-                                 "You have %d minute of battery power "
-				   "remaining (%d%% of the total capacity).",
-                                 "You have %d minutes of battery power "
-				   "remaining (%d%% of the total capacity).",
-                                 info->minutes ),
-                               info->minutes,info->percent );
-  }
-
-  if( is_suspend_unavailable() )
+    const char *suggest;
+    gchar *remaining, *new_label;
+    GtkRequisition size;
+
+    /* If we're not displaying the dialog then don't update it. */
+    if (battstat->battery_low_label == NULL ||
+        battstat->battery_low_dialog == NULL)
+        return;
+
+    gtk_widget_get_preferred_size (GTK_WIDGET (battstat->battery_low_label), NULL, &size);
+
+    /* If the label has never been set before, the width will be 0.  If it
+       has been set before (width > 0) then we want to keep the size of
+       the old widget (to keep the dialog from changing sizes) so we set it
+       explicitly here.
+     */
+    if (size.width > 0)
+        gtk_widget_set_size_request (GTK_WIDGET (battstat->battery_low_label),
+                                     size.width, size.height);
+
+    if (info->minutes < 0 && !info->on_ac_power)
+    {
+        /* we don't know the remaining time */
+        remaining = g_strdup_printf (_("You have %d%% of your total battery"
+                                     "capacity remaining."), info->percent);
+    }
+    else
+    {
+      remaining = g_strdup_printf (ngettext (
+                                   "You have %d minute of battery power "
+                                   "remaining (%d%% of the total capacity).",
+                                   "You have %d minutes of battery power "
+                                   "remaining (%d%% of the total capacity).",
+                                   info->minutes),
+                                   info->minutes,info->percent);
+    }
+
+    if (is_suspend_unavailable ())
     /* TRANSLATORS: this is a list, it is left as a single string
      * to allow you to make it appear like a list would in your
      * locale.  This is if the laptop does not support suspend. */
-    suggest = _("To avoid losing your work:\n"
-                " \xE2\x80\xA2 plug your laptop into external power, or\n"
-                " \xE2\x80\xA2 save open documents and shut your laptop down."
-               );
-  else
+        suggest = _("To avoid losing your work:\n"
+                  " \xE2\x80\xA2 plug your laptop into external power, or\n"
+                  " \xE2\x80\xA2 save open documents and shut your laptop down."
+                  );
```

Comment:
indent

Reply:
How?
After the bracket of the top line?
I wasn't sure with that one liner.

Reply:
```diff
diff --git a/battstat/battstat_applet.c b/battstat/battstat_applet.c
index 965331d0..b28555d5 100644
--- a/battstat/battstat_applet.c
+++ b/battstat/battstat_applet.c
@@ -376,17 +376,17 @@ battery_low_update_text (ProgressData  *battstat,
     {
         /* we don't know the remaining time */
         remaining = g_strdup_printf (_("You have %d%% of your total battery"
-                                     "capacity remaining."), info->percent);
+                                       "capacity remaining."), info->percent);
     }
     else
     {
-      remaining = g_strdup_printf (ngettext (
-                                   "You have %d minute of battery power "
-                                   "remaining (%d%% of the total capacity).",
-                                   "You have %d minutes of battery power "
-                                   "remaining (%d%% of the total capacity).",
-                                   info->minutes),
-                                   info->minutes,info->percent);
+      remaining = g_strdup_printf (ngettext ("You have %d minute of battery power "
+                                             "remaining (%d%% of the total capacity).",
+                                             "You have %d minutes of battery power "
+                                             "remaining (%d%% of the total capacity).",
+                                             info->minutes),
+                                   info->minutes,
+                                   info->percent);
     }
 
     if (is_suspend_unavailable ())
@@ -394,18 +394,16 @@ battery_low_update_text (ProgressData  *battstat,
      * to allow you to make it appear like a list would in your
      * locale.  This is if the laptop does not support suspend. */
         suggest = _("To avoid losing your work:\n"
-                  " \xE2\x80\xA2 plug your laptop into external power, or\n"
-                  " \xE2\x80\xA2 save open documents and shut your laptop down."
-                  );
+                    " \xE2\x80\xA2 plug your laptop into external power, or\n"
+                    " \xE2\x80\xA2 save open documents and shut your laptop down.");
     else
     /* TRANSLATORS: this is a list, it is left as a single string
      * to allow you to make it appear like a list would in your
      * locale.  This is if the laptop supports suspend. */
         suggest = _("To avoid losing your work:\n"
-                  " \xE2\x80\xA2 suspend your laptop to save power,\n"
-                  " \xE2\x80\xA2 plug your laptop into external power, or\n"
-                  " \xE2\x80\xA2 save open documents and shut your laptop down."
-                  );
+                    " \xE2\x80\xA2 suspend your laptop to save power,\n"
+                    " \xE2\x80\xA2 plug your laptop into external power, or\n"
+                    " \xE2\x80\xA2 save open documents and shut your laptop down.");
 
     new_label = g_strdup_printf ("<span weight=\"bold\" size=\"larger\">%s</span>\n\n%s\n\n%s",
                                  _("Your battery is running low"), remaining, suggest);
```